crypto/provider_core.c

```diff
@@ -556,9 +563,25 @@ OSSL_PROVIDER *ossl_provider_new(OSSL_LIB_CTX *libctx, const char *name,
         template.init = init_function;
     }
 
+    if (params != NULL) {
+        int i;
+
+        for (i = 0; params[i].key != NULL; i++) {
+            if (params[i].data_type != OSSL_PARAM_UTF8_STRING)
+                continue;
+            if (ossl_provider_info_add_parameter(&template, params[i].key,
```

Comment:
So the parameters are additive to any parameters that have already been obtained from a config file?

Reply:
I'm not sure. I think no because template is a local variable here and doesn't deal with the config file.

Reply:
The `store->provinfo` code directly above checks the config information from the config file. In particular see the function `provider_conf_load` in the crypto/provider_conf.c file in the case that `activate == 0`: it prepopulates `store->provinfo` with config data for later use by `ossl_provider_new`.

See line 549 above which causes `template` to be populated with data from the config file.

The comment on line 544 is misleading since this is actually not just related to user added builtin providers.

Reply:
Thanks! Added the copying of the parameters.